lib/active_record/virtual_attributes/virtual_total.rb

```diff
@@ -77,20 +96,19 @@ def define_virtual_aggregate_attribute(name, relation, method_name, column, opti
         end
       end
 
-      def define_virtual_size_method(name, relation)
-        define_method(name) do
-          (has_attribute?(name) ? self[name] : send(relation).try(:size)) || 0
-        end
-      end
-
-      def define_virtual_aggregate_method(name, relation, method_name, column)
+      def define_virtual_aggregate_method(name, relation, column, ruby_method_name, arel_method_name = ruby_method_name)
         define_method(name) do
           if has_attribute?(name)
             self[name] || 0
           elsif (rel = send(relation)).loaded?
```

Comment:
huh, will this work on a virtual association that returns an array?
(sorry, scope creeping you)

you can use `virtual_has_many :books_with_authors` -- not the most optimized use case but should work

get the sum of an author virtual attribute for extra credit ;) -- though to be honest, that is probably not a good idea since we don't lazy load through virtual associations. Well, maybe we do but I'm not optimistic.